### PR TITLE
Update Omicron to latest propolis with InstanceSerialConsoleHelper

### PR DESCRIPTION
**Pull request by leftwo**
Update Omicron to get newer Crucible and Propolis, which also includes the changes to support
the new Propolis `InstanceSerialConsoleHelper`.

### CODE REVIEW

File: nexus/src/app/instance.rs [hunk 1]
```diff
@@ -1362,6 +1361,43 @@ impl super::Nexus {
...
+                let ip_addr = instance
+                    .runtime_state
+                    .propolis_ip
+                    .ok_or_else(|| {
+                        Error::internal_error(
+                            "instance's hypervisor IP address not found",
+                        )
+                    })?
+                    .ip();
+                Ok(SocketAddr::new(ip_addr, PROPOLIS_PORT))
```
Comment (leftwo) [inline nexus/src/app/instance.rs:1385]: I think this might not be the proper port?

File: nexus/src/app/instance.rs [hunk 2]
```diff
@@ -1362,6 +1361,43 @@ impl super::Nexus {
         }
     }
 
+    async fn propolis_addr_for_instance(
```
Comment (sunshowers) [inline nexus/src/app/instance.rs:1364]: Could `propolis_client_for_instance` below use `propolis_addr_for_instance` to avoid duplicating code?

**Review by sunshowers — COMMENTED** [review]
Looks good overall, though I wrote half of it so maybe someone else should review it ;)

**Comment by leftwo** [comment]
> Looks good overall, though I wrote half of it so maybe someone else should review it ;)

And, thank you for your half (or really more than half).  :)

File: nexus/src/app/instance.rs (hunk 2 above)
Comment (leftwo) [reply nexus/src/app/instance.rs:1364]: For sure this is ripe for a refactor.
I want to be sure that we are actually using the correct address:port for the serial console and once I verify
that I'll refactor this to something better.

File: Cargo.lock [hunk 3]
```diff
@@ -6853,8 +6854,7 @@ dependencies = [
  "schemars_derive",
  "serde",
  "serde_json",
- "uuid 0.8.2",
```
Comment (ahl) [inline Cargo.lock:6856]: woot!

File: nexus/src/app/instance.rs (hunk 1 above)
Comment (leftwo) [reply nexus/src/app/instance.rs:1385]: Ah good, thanks.  I was not sure if we required a different port here for the serial console stuff.

File: nexus/src/app/instance.rs (hunk 2 above)
Comment (leftwo) [reply nexus/src/app/instance.rs:1364]: Refactor done.

**Comment by leftwo** [comment]
This should be good to go now.
I did test that with these changes I can still access the serial console on an instance.

**Review by lifning — APPROVED** [review]

**Pull request merged**